ons-spark/spark-analysis/flags.ipynb

```diff
@@ -0,0 +1,205 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Flags in Spark\n",
+    "\n",
+    "A flag is a column that indicates whether a certain condition, such as a threshold value, is met. The flag is typically a 0 or 1 depending on whether the condition has been met.\n",
+    "\n",
+    "Our example will show how you can create a flag for age differences greater than a threshold wihtin a given group. \n",
+    "\n",
+    "## Creating an age-difference flag\n",
+    "\n",
+    "Given an `id` column and an `age` column we want to create a flag (0 or 1) to indicate whether the age difference between the two eldest entries within a given `id` is greater than a specified threshold value. \n",
```

Comment:
I'm a little confused by "between the two eldest entries within a given `id`" - I'm not sure what it's referring to.

Does each ID have multiple ages, and you want to check whether the two eldest ages differ by an amount greater than a specified threshold value?

You might just have taken this wording directly from the original notebook, but it's a little confusing so may be worth rewording it slightly.

Reply:
Changed the text slightly, also renamed the column `id` to `group` to make it clear that column  refers to a group, since `id` identity column, is normally used to uniquely identify a row